src/resolve/resolved-dns-stub.c

```diff
@@ -992,6 +992,12 @@ static void dns_stub_process_query(Manager *m, DnsStubListenerExtra *l, DnsStrea
                                   SD_RESOLVED_NO_SEARCH|
                                   (DNS_PACKET_DO(p) ? SD_RESOLVED_REQUIRE_PRIMARY : 0)|
                                   SD_RESOLVED_CLAMP_TTL);
+
+        /* Refuse query if there is -ENOSYS */
+        if (r == -ENOANO) {
+                return (void) dns_stub_send_failure(m, l, s, p, DNS_RCODE_REFUSED, false);
+        }
+
```

Comment:
similar handling for ENOANO should be added for dns_query_new() calls in src/resolve/resolved-bus.c and src/resolve/resolved-varlink.c, returning a recognizable error.

in the former add a local wrapper dns_query_new_for_bus() or so, that fills in a useful sd_bus_error structure in case the ENOANO thing is seen. Then switch all callers over from dns_query_new() to this new wrapper call. 

Similar for the varlink case.

(background: we answer lookup requests via three mechanisms: D-Bus IPC, Varlink IPC and the local stub). Right now the latter would return a clean REFUSED error, but the others should return clean, descriptive errors for this too.